### PR TITLE
New command flag: spack install --dependencies-only

### PR DESCRIPTION
**Pull request by citibeth**
This allows Spack to install just a package's dependencies, without installing the package itself.  Useful for non-standard packages, such as `modele`, which can't be installed with a simple `spack install`.

### CODE REVIEW

**Comment by tgamblin** [comment]
@citibeth: this has test errors.

**Comment by citibeth** [comment]
I just rebased, should be ready to merge.

File: lib/spack/spack/package.py [hunk 1]
```diff
@@ -868,7 +868,8 @@ def _resource_stage(self, resource):
     def do_install(self,
                    keep_prefix=False,
                    keep_stage=False,
-                   ignore_deps=False,
+                   install_deps=True,
+                   install_self=True,
```
Comment (alalazo) [inline lib/spack/spack/package.py]: Sorry if I am late at being picky here, but I find the API of `do_install` growing quite unintuitive. Now we may call a `do_install` that does no install at all, while the semantic needed for this PR could have been obtained directly in the `install` command traversing the DAG with or without root and calling an unmodified version of `do_install` from there.

**Comment by citibeth** [comment]
I think this is worth a larger discussion on another Issues thread or the
mailing list.

On Wed, Oct 5, 2016 at 2:52 PM, Massimiliano Culpo <notifications@github.com

**Pull request merged**